src/precompile.jl

```diff
@@ -1,73 +1,14 @@
-        @warnpcfail precompile(Core.kwfunc(GridLayout), (NamedTuple{(:bbox, :alignmode), Tuple{NTuple{4,Int}, Al}}, Type{GridLayout}))
-        if fbody !== nothing
-            @warnpcfail precompile(fbody, (Nothing, Nothing, Nothing, Nothing, Nothing, Al, Tuple{Bool, Bool}, Rect2f, Auto, Auto, Bool, Bool, Symbol, Symbol, Float64, Float64, Emptykwargs, Type{GridLayout}, Int, Int))
-        end
-    end
-    @warnpcfail precompile(Core.kwfunc(GridLayout), (NamedTuple{(:colsizes, :rowsizes), Tuple{Fixed, Relative}}, Type{GridLayout}, Int, Int))
-    @warnpcfail precompile(Core.kwfunc(GridLayout), (NamedTuple{(:addedcolgaps,), Tuple{Vector{Fixed}}}, Type{GridLayout}, Int, Int))
-
-    @warnpcfail precompile(Core.kwfunc(Mixed), (NamedTuple{(:left, :top), Tuple{Int, Int}}, Type{Mixed}))
-    for T in (Int, Float32)
-        @warnpcfail precompile(Outside, (T,))
-        @warnpcfail precompile(Outside, (T,T,T,T))
-    end
-    @warnpcfail precompile(align_to_bbox!, (GridLayout, Rect2f))
-    @warnpcfail precompile(update!, (GridLayout,))
-    for I in subtypes(Indexables), J in subtypes(Indexables)
-        @warnpcfail precompile(setindex!, (GridLayout, UnitRange{Int}, I, J))
-        @warnpcfail precompile(setindex!, (GridLayout, Any, I, J))
-    end
-    @warnpcfail precompile(insertrows!, (GridLayout, Int, Int))
-    @warnpcfail precompile(insertcols!, (GridLayout, Int, Int))
-    @warnpcfail precompile(gridnest!, (GridLayout, UnitRange{Int}, UnitRange{Int}))
-    @warnpcfail precompile(add_to_gridlayout!, (GridLayout, GridContent{GridLayout, GridLayout}))
-    @warnpcfail precompile(connect_layoutobservables!, (GridContent{GridLayout, GridLayout},))
-    @warnpcfail precompile(trim!, (GridLayout,))
-    @warnpcfail precompile(sizeobservable!, (Observable{Any}, Observable{Any}))
-    @warnpcfail precompile(_reportedsizeobservable, (Tuple{SizeAttribute,SizeAttribute}, Tuple{AutoSize,AutoSize}, AlignMode, RectSides{Float32}, Tuple{Bool,Bool}))
-    for T in subtypes(Side)
-        @warnpcfail precompile(bbox_for_solving_from_side, (RowCols{Vector{Float64}}, Rect2f, RowCols{Int}, T))
-    end
-    for S in (Left, Right, Top, Bottom)
-        @warnpcfail precompile(protrusion, (GridContent{GridLayout,GridLayout}, S))
-    end
-    @warnpcfail precompile(suggestedbboxobservable, (GridLayout,))
-    for VC in (Vector{Auto}, Vector{GapSize}, Vector{Fixed}, Vector{Relative}, Vector{ContentSize})
-        @warnpcfail precompile(convert_contentsizes, (Int, VC))
-        @warnpcfail precompile(==, (VC, VC))
-    end
-    @warnpcfail precompile(contents, (GridLayout,))
-    @warnpcfail precompile(filterenum, (Function, Type, Vector{ContentSize}))
-    @warnpcfail precompile(zcumsum, (Vector{Float64},))
-
-    # These don't work completely but they have partial success
-    @warnpcfail precompile(repr, (MIME{Symbol("text/plain")}, GridLayout))
+let
+    while true
```

Comment:
what does this while true do?

Reply:
Forces it to be compiled. Code at toplevel is often run in the interpreter, but we have a heuristic that says anything with a loop in it should be compiled because we know the interpreter is really slow and there's no way to know in general how many iterations the loop will have.

I'll add a comment about this.